spring-boot-samples/spring-boot-sample-data-geode/src/main/java/sample/data/geode/SampleDataGeodeApplication.java

```diff
@@ -0,0 +1,161 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample.data.geode;
+
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.data.gemfire.CacheFactoryBean;
+import org.springframework.data.gemfire.GemfireTemplate;
+import org.springframework.data.gemfire.PartitionedRegionFactoryBean;
+import org.springframework.data.gemfire.RegionAttributesFactoryBean;
+
+import com.gemstone.gemfire.cache.Cache;
+import com.gemstone.gemfire.cache.CacheLoader;
+import com.gemstone.gemfire.cache.CacheLoaderException;
+import com.gemstone.gemfire.cache.LoaderHelper;
+import com.gemstone.gemfire.cache.RegionAttributes;
+
+import sample.data.geode.service.Calculator;
+import sample.data.geode.service.factory.CalculatorFactory;
+
+/**
+ * SampleDataGeodeApplication class is a Spring Boot, Apache Geode peer cache application using Geode
+ * to perform and store mathematical calculations.
+ *
+ * @author John Blum
+ * @see org.springframework.boot.SpringApplication
+ * @see org.springframework.boot.autoconfigure.SpringBootApplication
+ * @see org.springframework.boot.context.properties.ConfigurationProperties
+ * @see org.springframework.context.ApplicationContext
+ * @see org.springframework.context.annotation.Bean
+ * @see org.springframework.data.gemfire.CacheFactoryBean
+ * @see org.springframework.data.gemfire.GemfireTemplate
+ * @see org.springframework.data.gemfire.PartitionedRegionFactoryBean
+ * @see com.gemstone.gemfire.cache.Cache
+ * @see com.gemstone.gemfire.cache.CacheLoader
+ * @see sample.data.geode.service.Calculator
+ * @see sample.data.geode.service.factory.CalculatorFactory
+ * @since 1.4.0
+ */
+@SpringBootApplication
+@ConfigurationProperties
+@SuppressWarnings("unused")
+public class SampleDataGeodeApplication {
+
+	protected static final String CALCULATIONS_REGION_NAME = "Calculations";
+
+	public static void main(final String[] args) {
+		SpringApplication.run(SampleDataGeodeApplication.class, args);
+	}
+
+	@Bean
+	Properties gemfireProperties(@Value("${sample.data.geode.log.level:config}") String logLevel) {
```

Comment:
So, I think I completely misunderstood the meaning and purpose of `@ConfigurationProperties`.  My bad!

I assumed it was used to enable something akin to a `PropertyPlaceholderConfigurer` (e.g. `PropertySourcesPlaceholderConfigurer`) to resolve property placeholders/configuration (in `@Value` annotations used within application code) from some external source, as one possible feature among others.  I was basing my knowledge on [this](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-external-config), to which now I definitely see I am incorrect.

Interestingly, my sample code does work as I expect and the property placeholder values (in the `@Value` annotations) are resolved when I specify them on the command-line (or my IDE) or not (thus using the default values).

I have never really tried [this](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#boot-features-external-config-typesafe-configuration-properties), but seems like a pretty slick way to manage configuration properties.

I will clean up my mess! ;-)